diff --git a/config.go b/config.go
--- a/config.go
+++ b/config.go
@@ -28,6 +28,7 @@ type BuckConfig struct {
 	DepsAttr               string `json:"deps_attr"`                 // custom attribute for dependencies e.g. "go_external_deps". Default: "deps"
 	OSConstraintPrefix     string `json:"os_constraint_prefix"`      // prefix for OS constraint targets in select(), e.g. "config//os:". Default: "" (raw os name)
 	ArchConstraintPrefix   string `json:"arch_constraint_prefix"`    // prefix for arch constraint targets in select(), e.g. "config//cpu:". Default: "" (raw arch name)
+	BuildfileName          string `json:"buildfile_name"`            // output filename for build files. Default: "BUCK"
 }

 // PlatformConfig mapping between goos/goarch and buck os/arch
@@ -70,6 +71,9 @@ func ensureDefaultValues(c *Config) {
 	if c.Buck.DepsAttr == "" {
 		c.Buck.DepsAttr = "deps"
 	}
+	if c.Buck.BuildfileName == "" {
+		c.Buck.BuildfileName = "BUCK"
+	}
 	if c.Buck.DepsTargetLabelPrefix == "" {
 		slog.Info("buck.deps_target_label_prefix is not set, dependencies will be formatted as raw import paths")
 	}
diff --git a/render.go b/render.go
--- a/render.go
+++ b/render.go
@@ -111,7 +111,7 @@ func renderBuckFiles(cfg *Config, thirdPartyDir string, buckTargets BuckTargets)
 			// Acquire semaphore before doing IO
 			sem <- struct{}{}
 			defer func() { <-sem }()

-			buckFile := filepath.Join(thirdPartyDir, "vendor", target.ImportPath, "BUCK")
+			buckFile := filepath.Join(thirdPartyDir, "vendor", target.ImportPath, cfg.Buck.BuildfileName)
 			f, err := os.Create(buckFile)
 			if err != nil {
 				return fmt.Errorf("can't create BUCK file: %w", err)
